src/main/scala/xiangshan/backend/issue/Dispatch2Iq.scala

```diff
@@ -139,6 +142,351 @@ abstract class Dispatch2IqImp(override val wrapper: Dispatch2Iq)(implicit p: Par
+    IQ3Enq1Select := Cat(false.B, false.B, true.B, false.B).asBools
+  }.otherwise {
+    IQ2Enq0Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ2Enq1Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ3Enq0Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ3Enq1Select := Cat(false.B, false.B, false.B, false.B).asBools
+  }
+  assert(PopCount(IQ2Enq0Select) < 2.U, "IQ2Enq0Select is not one hot")
+  assert(PopCount(IQ2Enq1Select) < 2.U, "IQ2Enq1Select is not one hot")
+  assert(PopCount(IQ3Enq0Select) < 2.U, "IQ3Enq0Select is not one hot")
+  assert(PopCount(IQ3Enq1Select) < 2.U, "IQ3Enq1Select is not one hot")
+  assert(PopCount(IQ2Enq0Select) < 2.U, "IQ2Enq0Select is not one hot")
+  assert(PopCount(IQ2Enq1Select) < 2.U, "IQ2Enq1Select is not one hot")
+  assert(PopCount(IQ3Enq0Select) < 2.U, "IQ3Enq0Select is not one hot")
+  assert(PopCount(IQ3Enq1Select) < 2.U, "IQ3Enq1Select is not one hot")
+  assert(PopCount(VecInit(IQ2Enq0Select(0), IQ2Enq1Select(0), IQ3Enq0Select(0), IQ3Enq1Select(0))) < 2.U, "Dq1Uop0Select is not one hot")
+  assert(PopCount(VecInit(IQ2Enq0Select(1), IQ2Enq1Select(1), IQ3Enq0Select(1), IQ3Enq1Select(1))) < 2.U, "Dq1Uop1Select is not one hot")
+  assert(PopCount(VecInit(IQ2Enq0Select(2), IQ2Enq1Select(2), IQ3Enq0Select(2), IQ3Enq1Select(2))) < 2.U, "Dq1Uop2Select is not one hot")
+  assert(PopCount(VecInit(IQ2Enq0Select(3), IQ2Enq1Select(3), IQ3Enq0Select(3), IQ3Enq1Select(3))) < 2.U, "Dq1Uop3Select is not one hot")
+  val dq1Block = Wire(Vec(uopsInDq1Num, Bool()))
+  if (IQ2FuCfgs.size == IQ3FuCfgs.size && IQ2FuCfgs.size == IQ23BothfuCfgs.size) {
+    dq1Block := Cat(false.B, false.B, false.B, false.B).asBools
+  }
+  else {
+    val fuTypes = uopsInDq1.map(_.bits.fuType)
+    val iqNotReady0 = (FuTypeOrR(fuTypes(0), IQ2OnlyFuTypes) && !uopsOutDq1.head.ready) || (FuTypeOrR(fuTypes(0), IQ3OnlyFuTypes) && !uopsOutDq1.last.ready)
+    val iqNotReady1 = (FuTypeOrR(fuTypes(1), IQ2OnlyFuTypes) && !uopsOutDq1.head.ready) || (FuTypeOrR(fuTypes(1), IQ3OnlyFuTypes) && !uopsOutDq1.last.ready)
+    val iqNotReady2 = (FuTypeOrR(fuTypes(2), IQ2OnlyFuTypes) && !uopsOutDq1.head.ready) || (FuTypeOrR(fuTypes(2), IQ3OnlyFuTypes) && !uopsOutDq1.last.ready)
+    val iqNotReady3 = (FuTypeOrR(fuTypes(3), IQ2OnlyFuTypes) && !uopsOutDq1.head.ready) || (FuTypeOrR(fuTypes(3), IQ3OnlyFuTypes) && !uopsOutDq1.last.ready)
+    val conflict2 = (IQ2Enq0Select(2) || IQ2Enq1Select(2) && FuTypeOrR(fuTypes(2), IQ3OnlyFuTypes)) || (IQ3Enq0Select(2) || IQ3Enq0Select(2) && FuTypeOrR(fuTypes(2), IQ2OnlyFuTypes))
+    val conflict3 = (IQ2Enq0Select(3) || IQ2Enq1Select(3) && FuTypeOrR(fuTypes(3), IQ3OnlyFuTypes)) || (IQ3Enq0Select(3) || IQ3Enq0Select(3) && FuTypeOrR(fuTypes(3), IQ2OnlyFuTypes))
+    dq1Block(0) := iqNotReady0
+    dq1Block(1) := dq1Block(0) || iqNotReady1
+    dq1Block(2) := dq1Block(0) || dq1Block(1) || conflict2
+    dq1Block(3) := dq1Block(0) || dq1Block(1) || dq1Block(2) || conflict3
```

Comment:
dp0Block(1) is already has dq0Block(0) || iqNotReady1, so we can assign like this.
```scala
dq0Block(2) := dq0Block(1) || conflict2
dq0Block(3) := dq0Block(2) || conflict3
```